src/coingecko/Coingecko.ts

```diff
@@ -82,30 +145,74 @@ export class Coingecko {
     currency = "usd",
     platform_id = "ethereum"
   ): Promise<CoinGeckoPrice[]> {
-    // Generate a unique set with no repeated. join the set with the required coingecko delimiter.
-    const contract_addresses = Array.from(new Set(addresses.filter((n) => n).values()));
+    const priceCache: { [addr: string]: CoinGeckoPrice } = this.getPriceCache(currency, platform_id);
```

Comment:
i feel like the generic cache should be its own class where you use a generic to set the data type and have a basic interface like `set`, `get`, `has`, `delete` that take an argument for age, ie `has(key,timestamp)` returns false  if the value stored either does not exist or is older than timestamp, true otherwise.  the key could just be  string concatenation of `${currency}.${platform_id}.${address}`, or you could maintain separate caches for currency/platform like you do now.

Reply:
This is a really good suggestion and will making adding caching to other data clients in the future easier. However, how much extra code will this add? I'm a bit worried about adding generalization but trading off for future work where we need to retrofit a new way of caching data to the existing `Cache` client that we create. 

@daywiss can you give a prototype of this proposed change? Also, @pxrl isn't as familiar with javascript specific things like generics so it'd be helpful to understand more what you mean